### PR TITLE
Mimetypes

### PR DESCRIPTION
**Pull request by bollwyvl**
This changes the wire format to reference code as an "input bundle", a mime-typed list of `{"cell_id": "abc1234", code: "hello world"}`-type objects

All annotations of a mimetype are performed, with the lines skewed to the requested cell. This handles two cases:
- non "native" language (e.g. `text/x-ipythongfm` in `text/x-ipython`) are not linted by the "host" language
- annotations are still only shown to the cell that requests them, but the whole document is available to complete forward-references of types, catching of unimported modules, etc.

### CODE REVIEW

File: packages/ipylintotype/src/ipylintotype/formatter.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        if "parent" in kwargs:
+            kwargs = {**kwargs["parent"]._trait_values, **kwargs}
+        super().__init__(*args, **kwargs)
+        self.init_comm()
+
+    def init_comm(self):
+        self.close()
+        self.current_comm = Comm(target_name=self.comm_name)
+        self.current_comm.on_msg(self.on_msg)
+
```
Comment (tonyfast) [inline packages/ipylintotype/src/ipylintotype/formatter.py:48]: We need to close the COMM when the shell restarts. In init_comm we should register an atexit hook.

	__import__('atexit').register(AnnotationFormatter.close)

Comment (bollwyvl) [inline packages/ipylintotype/src/ipylintotype/formatter.py:48]: @tonyfast are you proposing
```py
def init_comm(self):
    ...
    __import__('atexit').register(AnnotationFormatter.close)
```

**Review by tonyfast — APPROVED** [review]

File: packages/ipylintotype/src/ipylintotype/formatter.py (hunk 1 above)
Comment (tonyfast) [reply packages/ipylintotype/src/ipylintotype/formatter.py:48]: Exactly it belongs in the init_comm closure.

Comment (bollwyvl) [reply packages/ipylintotype/src/ipylintotype/formatter.py:48]: hit it in 56b7b11

**Comment by bollwyvl** [comment]
added some widgets

**Pull request merged**